src/GovUk.Education.ExploreEducationStatistics.Admin/Services/MetaGuidanceService.cs

```diff
@@ -0,0 +1,186 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Release = GovUk.Education.ExploreEducationStatistics.Content.Model.Release;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class MetaGuidanceService : IMetaGuidanceService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _contentPersistenceHelper;
+        private readonly IFilterService _filterService;
+        private readonly IIndicatorService _indicatorService;
+        private readonly StatisticsDbContext _statisticsDbContext;
+        private readonly IPersistenceHelper<StatisticsDbContext> _statisticsPersistenceHelper;
+        private readonly IUserService _userService;
+
+        public MetaGuidanceService(ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> contentPersistenceHelper,
+            IFilterService filterService,
+            IIndicatorService indicatorService,
+            StatisticsDbContext statisticsDbContext,
+            IPersistenceHelper<StatisticsDbContext> statisticsPersistenceHelper,
+            IUserService userService)
+        {
+            _contentDbContext = contentDbContext;
+            _contentPersistenceHelper = contentPersistenceHelper;
+            _filterService = filterService;
+            _indicatorService = indicatorService;
+            _statisticsDbContext = statisticsDbContext;
+            _statisticsPersistenceHelper = statisticsPersistenceHelper;
+            _userService = userService;
+        }
+
+        public async Task<Either<ActionResult, MetaGuidanceViewModel>> Get(Guid releaseId)
+        {
+            return await _contentPersistenceHelper.CheckEntityExists<Release>(releaseId)
+                .OnSuccessDo(release => _userService.CheckCanViewRelease(release))
+                .OnSuccess(async release => BuildViewModel(release, await GetSubjects(releaseId)));
+        }
+
+        public async Task<Either<ActionResult, MetaGuidanceViewModel>> UpdateRelease(Guid releaseId,
+            MetaGuidanceUpdateReleaseViewModel request)
+        {
+            return await _contentPersistenceHelper.CheckEntityExists<Release>(releaseId)
+                .OnSuccessDo(release => _userService.CheckCanUpdateRelease(release))
+                .OnSuccess(async release =>
+                {
+                    _contentDbContext.Update(release);
+                    release.MetaGuidance = request.Content;
+                    await _contentDbContext.SaveChangesAsync();
+
+                    return BuildViewModel(release, await GetSubjects(releaseId));
+                });
+        }
+
+        public async Task<Either<ActionResult, MetaGuidanceViewModel>> UpdateSubject(Guid releaseId,
+            Guid subjectId,
+            MetaGuidanceUpdateSubjectViewModel request)
+        {
+            return await _contentPersistenceHelper.CheckEntityExists<Release>(releaseId)
```

Comment:
Rather than two separate requests, we could do a single `CheckEntityExists<ReleaseSubject>` where we add an `Include` for the Release.

Reply:
I already thought of that, but the call proceeding directly after this, `_userService.CheckCanUpdateRelease` needs a Content Release whereas the Release of a ReleaseSubject is a Statistics Release.

Reply:
Ah yeh of course. Fine with it as it is then :+1: